File: admin/src/app/core/interceptors/auth.interceptor.ts
import { HttpInterceptorFn } from '@angular/common/http';
import { inject } from '@angular/core';

import { AuthService } from '../services/auth.service';

const PUBLIC_PATHS = ['/auth/login', '/auth/refresh'];

/**
 * @description
 * Adiciona o cabeçalho `Authorization: Bearer <token>` nas requisições de saída,
 * menos nos endpoints públicos de autenticação (login e refresh).
 */
export const authInterceptor: HttpInterceptorFn = (req, next) => {
  if (PUBLIC_PATHS.some((path) => req.url.includes(path))) {
    return next(req);
  }

  const auth = inject(AuthService);
  const token = auth.getAccessToken();
  if (!token) {
    return next(req);
  }

  const authorized = req.clone({
    setHeaders: { Authorization: `Bearer ${token}` },
  });
  return next(authorized);
};

File: mobile/lib/core/network/interceptors/auth_interceptor.dart
import 'package:conectaparana/core/auth/auth_service.dart';
import 'package:dio/dio.dart';

class AuthInterceptor extends Interceptor {
  final AuthService _authService = AuthService.instance;

  @override
  void onRequest(
    RequestOptions options,
    RequestInterceptorHandler handler,
  ) async {
    final requiredAuth = options.extra['auth'] == true;

    if (!requiredAuth) return handler.next(options);

    final token = await _authService.getAccessToken();

    if (token != null) options.headers['Authorization'] = 'Bearer $token';

    handler.next(options);
  }
}
